pytest_html/extras.py

```diff
@@ -9,16 +9,32 @@
 FORMAT_URL = 'url'
 
 
-def extra(content, format, name=None):
-    return {'name': name, 'format': format, 'content': content}
+def extra(content, format, name=None, mime_type='image/png', extension='png'):
+    return {'name': name, 'format': format, 'content': content,
+            'mime_type': mime_type, 'extension': extension}
 
 
 def html(content):
     return extra(content, FORMAT_HTML)
 
 
 def image(content, name='Image'):
-    return extra(content, FORMAT_IMAGE, name)
+    return png(content=content, name=name)
+
+
+def png(content, name='Image'):
+    return extra(content, FORMAT_IMAGE, name,
+                 mime_type='image/png', extension='png')
+
+
+def jpg(content, name='Image'):
+    return extra(content, FORMAT_IMAGE, name,
+                 mime_type='image/jpeg', extension='jpg')
+
+
+def svg(content, name='Image'):
+    return extra(content, FORMAT_IMAGE, name,
+                 mime_type='image/svg+xml', extension='svg')
 
 
 def json(content, name='JSON'):
```

Comment:
```python
return extra(content, FORMAT_JSON, name, 'text/json', 'json')
```

Then you can use `extra.get('mime_type')` and `extra.get('extension')` in the block starting on line 184 of plugin.py. Same applies to `text`.